### PR TITLE
Fix ld issue on go 1.23

### PR DESCRIPTION
**Pull request by brandonpollack23**
As per pgavlin/text#1, a workaround for this so go 1.23 works is this ld
flag -checklintname=0 until they get around to fixing it.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -65,21 +65,21 @@ generate::
 	echo "This command does not do anything anymore. It will be removed in a future version."
 
 build:: build_proto build_display_wasm go.ensure
-	cd pkg && go install -ldflags "-X github.com/pulumi/pulumi/pkg/v3/version.Version=${VERSION}" ${PROJECT}
+	cd pkg && go install -ldflags "-X github.com/pulumi/pulumi/pkg/v3/version.Version=${VERSION} -checklinkname=0" ${PROJECT}
```
[the remark this answers is not in the document]
Comment (brandonpollack23) [reply Makefile:68]: That seems like a good idea to me ill add one called EXTRA_LD_FLAGS or some such

File: Makefile [hunk 2]
```diff
@@ -19,6 +19,9 @@ LINT_GOLANG_PKGS := sdk pkg tests sdk/go/pulumi-language-go sdk/nodejs/cmd/pulum
 # Additional arguments to pass to golangci-lint.
 GOLANGCI_LINT_ARGS ?=
 
+# Additional flags to pass to the go compiler ldflags
```
Comment (tgummerer) [inline Makefile:22]:
```suggestion
# Additional flags to pass to the go compiler ldflags
#
# TODO: revert this once [pgavlin/text#1] is fixed.
```

**Review by tgummerer — COMMENTED** [review]
Thanks, this looks like a reasonable shortterm workaround.  

We should also update CI to run Go 1.23 to actually test this.  I don't think this fix is complete yet, `goreleaser` uses `scripts/go-wrapper.sh` to build the binaries iirc, so at least that will need to be updated as well (or maybe we should switch that to use the Makefile to be less confusing, not sure how easy that would be). 

It might also be worth having a look if we can get rid of this dependency.

**Comment by julienp** [comment]
> We should also update CI to run Go 1.23 to actually test this. I don't think this fix is complete yet, `goreleaser` uses `scripts/go-wrapper.sh` to build the binaries iirc, so at least that will need to be updated as well (or maybe we should switch that to use the Makefile to be less confusing, not sure how easy that would be).

I think you can set `ldflags` as a property in `.goreleaser.yml`, should we add it there for each binary, or in the wrapper script?

The CI version can be updated here https://github.com/pulumi/pulumi/blob/d504bef75cbbf756a5f7277340743e43297c4ae6/scripts/get-job-matrix.py#L131

**Comment by tgummerer** [comment]
> I think you can set `ldflags` as a property in `.goreleaser.yml`, should we add it there for each binary, or in the wrapper script?

I'm not sure what the best way to do it is tbh.  Either works for me :)

**Comment by brandonpollack23** [comment]
just pushed some changes let me know if there is anything else needed i think i got it all

**Comment by tgummerer** [comment]
Uh fun, so the version of `golangci-lint` we're using here seems to be broken with Go 1.23 (even trying it out locally it eventually gets killed, so it's not just slightly out of memory on GitHub actions or anything).  Now upgrading to the latest golangci-lint version seems to fix that, but also raises a bunch of new warnings.  So unfortunately I think for Go 1.23 we need a bit more work and upgrade `golangci-lint` as well and fix those warnings.

**Comment by justinvp** [comment]
FYI, Pat fixed the pgavlin/text issue, so we can remove the workaround; see #17041

**Comment by brandonpollack23** [comment]
alright perhaps abandoning this change is ok until we figure that out?

**Comment by tgummerer** [comment]
> alright perhaps abandoning this change is ok until we figure that out?

Yeah I think we can abandon this.  We still need to update the dependency and upgrade golangci-lint in case you're interested in working on those :)

**Pull request closed without merging**